Source/WebCore/platform/graphics/gbm/DMABufFormat.h

```diff
@@ -129,6 +133,15 @@ struct DMABufFormat {
         FourCC fourcc { FourCC::Invalid };
         unsigned horizontalSubsampling { 0 };
         unsigned verticalSubsampling { 0 };
+
+    private:
+        Plane(const FourCC& fourcc, const unsigned& hsValue, const unsigned& vsValue)
+            : fourcc(fourcc)
+            , horizontalSubsampling(hsValue)
+            , verticalSubsampling(vsValue)
+        { }
+
+        friend struct IPC::ArgumentCoder<Plane, void>;
```

Comment:
You shouldn't need to add an explicit constructor since this is a struct and the data members are public, so aggregate initialization should work. If you use the same order in the serialization declaration as they are declared in the struct, the generated code should be able to create a new instance with something like `Plane { WTFMove(arg1), WTFMove(arg2), WTFMove(arg3) }`.

Reply:
AFAIK, in c++, structs can indeed benefit from aggregate initialization if they meet certain conditions like **no user-provided or inherited constructors**, no private or protected non-static data members, no virtual functions, and no virtual, private, or protected base classes.

However, the struct Plane does not qualify for aggregate initialization because it includes a user-provided constructor.

``` cpp
    struct Plane {                                                                                                     
        Plane() = default;                                                                                             
    
        template<typename PlaneDefinitionType>                                                                         
        Plane(const PlaneDefinitionType&)                                                                              
            : fourcc(PlaneDefinitionType::fourcc)                                                                      
            , horizontalSubsampling(PlaneDefinitionType::horizontalSubsampling)                                        
            , verticalSubsampling(PlaneDefinitionType::verticalSubsampling)                                            
        { }   
...
```